DigitalLearningSolutions.Web.Tests/Helpers/RegistrationEmailValidatorTests.cs

```diff
@@ -0,0 +1,346 @@
+                model,
+                modelState,
+                userService,
+                centresService
+            );
+
+            // Then
+            A.CallTo(() => userService.EmailIsInUse(DefaultPrimaryEmail)).MustHaveHappenedOnceExactly();
+            A.CallTo(() => centresService.DoesEmailMatchCentre(DefaultPrimaryEmail, DefaultCentreId))
+                .MustHaveHappenedOnceExactly();
+            modelState[nameof(PersonalInformationViewModel.PrimaryEmail)].ValidationState.Should()
+                .Be(ModelValidationState.Invalid);
+            AssertModelStateErrorIsExpected(
+                nameof(PersonalInformationViewModel.PrimaryEmail),
+                WrongEmailForCentreErrorMessage
+            );
+        }
+
+        [Test]
+        public void
+            ValidateEmailAddressesForAdminRegistration_adds_wrong_email_error_on_secondary_if_neither_email_matches_centre_and_secondary_is_not_null()
+        {
+            // Given
+            var model = GetDefaultPersonalInformationViewModel();
+
+            A.CallTo(() => userService.EmailIsInUse(DefaultPrimaryEmail)).Returns(false);
+            A.CallTo(() => userService.EmailIsInUse(DefaultSecondaryEmail)).Returns(false);
+            A.CallTo(
+                () => centresService.DoesEmailMatchCentre(
+                    DefaultPrimaryEmail,
+                    DefaultCentreId
+                )
+            ).Returns(false);
+            A.CallTo(
+                () => centresService.DoesEmailMatchCentre(
+                    DefaultSecondaryEmail,
+                    DefaultCentreId
+                )
+            ).Returns(false);
+
+            // When
+            RegistrationEmailValidator.ValidateEmailAddressesForAdminRegistration(
+                model,
+                modelState,
+                userService,
+                centresService
+            );
+
+            // Then
+            A.CallTo(() => userService.EmailIsInUse(DefaultPrimaryEmail)).MustHaveHappenedOnceExactly();
+            A.CallTo(() => userService.EmailIsInUse(DefaultSecondaryEmail)).MustHaveHappenedOnceExactly();
+            A.CallTo(() => centresService.DoesEmailMatchCentre(DefaultPrimaryEmail, DefaultCentreId))
+                .MustHaveHappenedOnceExactly();
+            A.CallTo(() => centresService.DoesEmailMatchCentre(DefaultSecondaryEmail, DefaultCentreId))
+                .MustHaveHappenedOnceExactly();
+            modelState[nameof(PersonalInformationViewModel.SecondaryEmail)].ValidationState.Should()
+                .Be(ModelValidationState.Invalid);
+            AssertModelStateErrorIsExpected(
+                nameof(PersonalInformationViewModel.SecondaryEmail),
+                WrongEmailForCentreErrorMessage
+            );
+        }
+
+        [Test]
+        public void
+            ValidateEmailAddressesForAdminRegistration_does_not_add_wrong_email_error_if_primary_is_unique_and_matches_centre_and_secondary_is_null()

```

Comment:
Can we have a unit test to make sure we don't have any errors added on when both primary and secondary emails are supplied, are unique, and only the primary email matches the centre?

Reply:
Yup, added one :)